### ISSUE TITLE
Deployment Hangs if Strategy Specified

### ISSUE THREAD

**Issue by tdickman** [kubernetes/kubernetes#23677]
If I run the following deployment, it hangs with 0 deployed nodes.

```
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  name: kub-hello-world-deployment
spec:
  replicas: 2
  strategy:
    type: rollingUpdate
  template:
    metadata:
      labels:
        app: kub-hello-world
        build: build24
    spec:
      containers:
      - name: kub-hello-world
        image: gcr.io/labs-sandbox/kub-hello-world:build24
        ports:
        - containerPort: 5000
```

`kubectl get deployments`

```
NAME                         DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
kub-hello-world-deployment   2         0         0            0           24m
```

If I run the same deployment without the spec.strategy.type, the deployment occurs correctly with no issues, and 2 nodes are created.

This is occurring on Kubernetes 1.2.0 on Google Cloud.

**Comment by tdickman** [comment]
After further debugging, it looks like spec.strategy.type should start with a capital letter. (RollingUpdate vs rollingUpdate). It seems like this should be rejected by the template processor rather than failing silently.

**Comment by zhouhaibing089** [comment]
This code may help a little bit to explain:

```go
func ValidateDeploymentStrategy(strategy *extensions.DeploymentStrategy, fldPath *field.Path) field.ErrorList {
	allErrs := field.ErrorList{}
	if strategy.RollingUpdate == nil {
		return allErrs
	}
	switch strategy.Type {
	case extensions.RecreateDeploymentStrategyType:
		allErrs = append(allErrs, field.Forbidden(fldPath.Child("rollingUpdate"), "may not be specified when strategy `type` is '"+string(extensions.RecreateDeploymentStrategyType+"'")))
	case extensions.RollingUpdateDeploymentStrategyType:
		allErrs = append(allErrs, ValidateRollingUpdateDeployment(strategy.RollingUpdate, fldPath.Child("rollingUpdate"))...)
	}
	return allErrs
}
```

**Comment by janetkuo** [comment]
There's a fix #27160 ready
